### PR TITLE
Add generic omniauth oauth2 suport

### PR DESCRIPTION
**Pull request by Stel000**
We wanted Single Sign On to our services and Gitlab for users and also need deploy quick. I refer the Document of Gitlab and add this feature.It allow user to configure the environment to enable generic omniauth oauth2.

### CODE REVIEW

**Comment by solidnerd** [comment]
@Stel000 Thanks for your Contribution. Is there a way how can someone test it ? With an example omniauth service. That we can ensure that it's working correctly.

**Comment by Stel000** [comment]
@solidnerd I'm sorry i can't.My employer's plantform use too more weak password.

**Comment by carllhw** [comment]
Any progress ?

File: README.md [hunk 1]
```diff
@@ -1018,6 +1026,24 @@ Below is the complete list of available options that can be used to customize yo
 | `OAUTH_AZURE_API_KEY` | Azure Client ID. No defaults. |
 | `OAUTH_AZURE_API_SECRET` | Azure Client secret. No defaults. |
 | `OAUTH_AZURE_TENANT_ID` | Azure Tenant ID. No defaults. |
+| `OAUTH_GENERIC_CLIENT_ID` | Your own OAuth2 Client ID. No defaults. |
+| `OAUTH_GENERIC_CLIENT_SECRET` | Your own OAuth2 Client secret. No defaults. |
+| `OAUTH_GENERIC_SITE` | The URL for your OAuth 2 server(eg. https://your_oauth_server).Including port if necessary. No defaults. |
+| `OAUTH_GENERIC_USER_INFO_URL` | The endpoint on your OAuth 2 server that provides user info for the current user. No defaults. |
```
Comment (raphaelyancey) [inline README.md]: Do you need a path (then the URL is built with OAUTH_GENERIC_SITE) or the full URL with host?

**Pull request closed without merging**